src/gc/windows/gcenv.windows.cpp

```diff
@@ -699,6 +737,31 @@ bool GCToOSInterface::VirtualRelease(void* address, size_t size)
     return !!::VirtualFree(address, 0, MEM_RELEASE);
 }
 
+// Commit virtual memory range.
+// Parameters:
+//  size      - size of the virtual memory range
+// Return:
+//  Starting virtual address of the committed range
+void* GCToOSInterface::VirtualReserveAndCommitLargePages(size_t size)
+{
+    void* pRetVal = nullptr;
+
+    if (!g_SeLockMemoryPrivilegeAcquired)
+    {
+        if (!InitLargePagesPrivilege())
+        {
+            return nullptr;
+        }
+
+        g_SeLockMemoryPrivilegeAcquired = true;
+    }
+
+    SIZE_T largePageMinimum = GetLargePageMinimum();
```

Comment:
Do we really have to call GetLargePageMinimum() every time we allocate VA or this value can be acquired once and cached in the init function?

Reply:
This function is only called once (or twice in the case split blocks) at startup. I’m not sure it’s worth putting another static for this but I certainly don’t feel strongly one way or the other.

Reply:
This function is only called once (or twice in the case split blocks) at startup. I’m not sure it’s worth putting another static for this but I certainly don’t feel strongly one way or the other.